open_bounties/0077 JavaScript Code Samples 2.md

```diff
@@ -0,0 +1,40 @@
+---
+ID: "0077"
+Discussion: https://github.com/XRPLBounties/Proposals/discussions/77
+Category: Technical Documentation
+Status: Open
+Amount: $8250
+---
+
+# JavaScript Code Samples 2
+
+## Overview
+
+Currently, our JavaScript documentations have [code snippets](https://github.com/XRPLF/xrpl.js/tree/main/packages/xrpl/snippets/src) and [code samples](https://github.com/XRPLF/xrpl-dev-portal/tree/master/content/_code-samples) for some common usecases. However, there needs to be more examples to capture common tasks that users would want the library to do, and fixes to outdated or misconfigured snippets. Developers would benefit from additional relevant, high quality, well-documented examples in JavaScript.
+
+## Details
+
+To add missing code samples, follow the instructions at the bottom of the page, under **Contribute Code Samples**.
```

Comment:
I thought this is to add to [code snippets](https://github.com/XRPLF/xrpl.js/tree/main/packages/xrpl/snippets/src)?

Reply:
Some of the changes should be in xrpl-dev-portal (like changing the depreciated codes), so my initial thought would be to have all of them in xrpl-dev-portal first then move them around later if needed. We can also add a column (Sample/Snippet) to point each of them to a location we want, but the difference between snippets in xrpl-js and code sample in xrpl.org is kinda slim so I'm not sure. @mDuo13 what do you think?

Reply:
Makes sense to me. Loosely, I think I originally pictured the dev portal list of code samples as being bigger / more complete snippets or examples of how to use the ledger in general, while the in-repo snippets were smaller bits or more specific to the library. In practice, I'm now of the opinion that that distinction is too fuzzy to be useful, and it's fine to put all of them in the dev portal.